product/views.py

```diff
@@ -54,16 +65,19 @@ def get(self,request):
                     'price'           : product.price,
                     'created_time'    : product.created_at,
                     'product_image'   : product.image_set.get().image_url,
-                    'sale_amount'     : 10,
-                    } for product in products]
+                    'discount'        : product.discount.rate,
+                    'stock'           : product.is_in_stock,
+                    } for product in products[offset:(limit+offset)]]
 
         return JsonResponse({'PRODUCTS': all_product}, status=200)
 
 class ProductDetailView(View):
-    def get(self, request, product_id):
+    def get(self, request):
         try:
-            product = Product.objects.get(id=product_id)
-        
+            product_id = int(request.GET.get('product', 1))
```

Comment:
Restful API를 구현할 때, `id = 1`인  product를 불러와야 하는 경우 `path parameter`를 사용해서 구현해 주셔야 합니다.
```suggestion
    def get(self, request, product_id):

```

Reply:
path parameter로 구현하였습니다.
우선 하나의 클래스로 id로 접근하는 방법과 search String으로 접근하는것을 모두 구현하였습니다
이같이 2가지를 모두 사용하기 위해서 URL이 아래와 같아지는데 이 방법이 맞는건지 궁금합니다.
10.XXX.XXX.XX:8000/product/1 --> product_id로 찾기
10.XXX.XXX.XX:8000/product/1?search='str' --> product_name으로 찾기
위와 같이 urls.py를 <int:product_id>로 수정하여 반드시 숫자를 입력한 후 쿼리스트링을 시작해야합니다.
의견 부탁드립니다.